File: pset1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>
int main()
{
    float cash;
    int cents; // cents 100 * cash
    int num_coins = 0; // answer
    do
    {
        
        cash = get_float("Change owed: "); // ask for change in float
   
    }
    while (cash < 0);
    
    cents = round(cash * 100); // cents = 100 * cash
    num_coins += cents / 25 - (cents / 25) % 1; // use as many quarters
    cents = cents % 25; // remaining cash
    num_coins += cents / 10 - (cents / 10) % 1; // use as many dimes
    cents = cents % 10; // remaining cash
    num_coins += cents / 5 - (cents / 5) % 1; // use as many nickels
    cents = cents % 5; // remaining cash
    num_coins += cents / 1; // use as many pennies
    // no remaining cash
    printf("%d\n", num_coins);
}

File: pset6/cash/cash.py
from cs50 import get_float
from math import floor


def main():
    while True:  # while cents_owed < 0 or string
        dollars_owed = get_float('Change owed: ')
        cents_owed = floor(dollars_owed * 100)

        if cents_owed > 0:
            break

    quarters = cents_owed // 25  # cents divide by 25
    dimes = (cents_owed % 25) // 10  # remaining cents divide by 10
    nickles = ((cents_owed % 25) % 10) // 5  # remaining cents divide by 5
    pennies = (((cents_owed % 25) % 10) % 5)  # remaining cents

    print(f"{quarters + dimes + nickles + pennies}")  # printing result


if __name__ == "__main__":
    main()